util/codec/bytes.go

```diff
@@ -122,6 +122,27 @@ func DecodeBytesDesc(b []byte) ([]byte, []byte, error) {
 	return decodeBytes(b, true)
 }
 
+// EncodeCompactBytes joins bytes with its length into a byte slice. It is more
+// efficient in both space and time compare to EncodeBytes.
+func EncodeCompactBytes(b []byte, data []byte) []byte {
+	b = EncodeVarint(b, int64(len(data)))
```

Comment:
Why using 9 here? I suppose we should use `binary.MaxVarintLen64`, which is 10.
As we would rarely deal with very long strings, maybe `binary.MaxVarintLen16` is enough.